gensim/test/test_similarities.py

```diff
@@ -427,6 +427,17 @@ def testReopen(self):
         self.assertTrue(numpy.allclose(expected, sims))
         index.destroy()
 
+    def testShardDir(self):
+        """test where shard and pickles are moved """
+        
+        homedir = os.path.join(os.path.expanduser('~'),'')
+        index = similarities.Similarity(homedir, corpus[:5], num_features=len(dictionary), shardsize=9)
+        index.save('test_index')
+        self.assertTrue(os.path.exists(os.path.join(os.path.dirname(homedir), 'shard', '')))
+        self.assertTrue(os.path.exists(os.path.join(os.path.dirname(homedir), 'shard', '.0')))
+        self.assertTrue(os.path.exists(os.path.join(os.path.dirname(homedir), 'shard', 'test_index')))
+        index.destroy()
```

Comment:
yes, things to do 
- [x] create shard directory, and test for pickle and shard
- [ ] move the files with os.rename
- [ ] load the files from new location, and test for check_moved()
- [ ] delete shard directory (destroy workspace thus created)

I'll do it asap.Thanks @tmylk